### ISSUE TITLE
Allow throwing errors in callsFake() mock

### ISSUE THREAD

**Issue by m-radzikowski** [m-radzikowski/aws-sdk-client-mock#8]
Currently, the `callsFake()` mock does not support throwing an error from it. The error should be returned as a Client `send()` error, in the same way as `mock.rejects()` behaves.

-----

Please vote with 👍 &nbsp; if you need this.

**Comment by bgardner-noggin** [comment]
I managed to get this to work for my code


```
let getParameters;
iamMock.on(GetSAMLProviderCommand.callsFake((input) => {
 getParameters = input;
 throw Error('forcing error');
});
...
expect(getParameters.Name).toEqual('Foo');
```

**Comment by m-radzikowski** [comment]
@bgardner-noggin this throws an error from the test that is not caught by jest, so this test fails with an error:

```ts
it('throws an error', async () => {
    snsMock.on(PublishCommand).callsFake(() => {
        throw new Error('forcing error');
    });

    const sns = new SNSClient({});

    await expect(sns.send(publishCmd1)).rejects.toThrow('forcing error');
});
```

However, this nudged me to look into handling errors from `callsFake()`, and I've just added it and released as v2.2.0.
